File: flutter_app/lib/widgets/step_progress_bar.dart
import 'package:flutter/material.dart';

import '../theme/colors.dart';

/// Port de `ui/components/StepProgressBar.kt`.
const List<String> stepLabels = [
  'Identité',
  'Expériences',
  'Formation',
  'Compétences',
  'Aperçu',
];

class StepProgressBar extends StatelessWidget {
  const StepProgressBar({
    super.key,
    required this.currentStep,
    required this.totalSteps,
    required this.onStepClick,
  });

  final int currentStep;
  final int totalSteps;
  final ValueChanged<int> onStepClick;

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;
    final children = <Widget>[];

    for (var index = 0; index < totalSteps; index++) {
      final isCompleted = index < currentStep;
      final isCurrent = index == currentStep;
      final color =
          (isCompleted || isCurrent) ? AppColors.gold : AppColors.mediumGray;

      children.add(
        Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            GestureDetector(
              onTap: () => onStepClick(index),
              child: Container(
                width: 32,
                height: 32,
                alignment: Alignment.center,
                decoration: BoxDecoration(color: color, shape: BoxShape.circle),
                child: Text(
                  isCompleted ? '✓' : '${index + 1}',
                  style: TextStyle(
                    fontSize: 12,
                    fontWeight: FontWeight.bold,
                    color: (isCompleted || isCurrent)
                        ? scheme.onPrimary
                        : scheme.onSurfaceVariant,
                  ),
                ),
              ),
            ),
            const SizedBox(height: 4),
            Text(
              index < stepLabels.length ? stepLabels[index] : '',
              style: TextStyle(
                fontSize: 9,
                color: isCurrent ? AppColors.gold : scheme.onSurfaceVariant,
                fontWeight: isCurrent ? FontWeight.w600 : FontWeight.normal,
              ),
            ),
          ],
        ),
      );

      if (index < totalSteps - 1) {
        children.add(
          Expanded(
            child: Padding(
              padding: const EdgeInsets.only(bottom: 16),
              child: SizedBox(
                height: 2,
                child: Stack(
                  children: [
                    const Positioned.fill(
                      child: ColoredBox(color: AppColors.mediumGray),
                    ),
                    AnimatedFractionallySizedBox(
                      duration: const Duration(milliseconds: 300),
                      widthFactor: index < currentStep ? 1 : 0,
                      alignment: Alignment.centerLeft,
                      child: const ColoredBox(color: AppColors.gold),
                    ),
                  ],
                ),
              ),
            ),
          ),
        );
      }
    }

    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 12),
      child: Row(
        crossAxisAlignment: CrossAxisAlignment.center,
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: children,
      ),
    );
  }
}

File: app/src/main/java/com/cvgenerator/ui/components/StepProgressBar.kt
package com.cvgenerator.ui.components

import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.animation.core.tween
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.cvgenerator.ui.theme.Gold
import com.cvgenerator.ui.theme.MediumGray

val stepLabels = listOf("Identité", "Expériences", "Formation", "Compétences", "Aperçu")

@Composable
fun StepProgressBar(
    currentStep: Int,
    totalSteps: Int,
    onStepClick: (Int) -> Unit,
    modifier: Modifier = Modifier
) {
    Column(modifier = modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 12.dp)) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            repeat(totalSteps) { index ->
                val isCompleted = index < currentStep
                val isCurrent = index == currentStep
                val color = when {
                    isCompleted || isCurrent -> Gold
                    else -> MediumGray
                }

                Column(horizontalAlignment = Alignment.CenterHorizontally) {
                    Box(
                        modifier = Modifier
                            .size(32.dp)
                            .clip(CircleShape)
                            .background(color)
                            .clickable { onStepClick(index) },
                        contentAlignment = Alignment.Center
                    ) {
                        Text(
                            text = if (isCompleted) "✓" else "${index + 1}",
                            color = if (isCompleted || isCurrent) MaterialTheme.colorScheme.onPrimary
                            else MaterialTheme.colorScheme.onSurfaceVariant,
                            fontSize = 12.sp,
                            fontWeight = FontWeight.Bold
                        )
                    }
                    Spacer(Modifier.height(4.dp))
                    Text(
                        text = stepLabels.getOrElse(index) { "" },
                        fontSize = 9.sp,
                        color = if (isCurrent) Gold else MaterialTheme.colorScheme.onSurfaceVariant,
                        fontWeight = if (isCurrent) FontWeight.SemiBold else FontWeight.Normal
                    )
                }

                if (index < totalSteps - 1) {
                    val progress by animateFloatAsState(
                        targetValue = if (index < currentStep) 1f else 0f,
                        animationSpec = tween(300), label = "progress"
                    )
                    Box(
                        modifier = Modifier
                            .weight(1f)
                            .height(2.dp)
                            .padding(bottom = 16.dp)
                    ) {
                        Box(modifier = Modifier.fillMaxSize().background(MediumGray))
                        Box(
                            modifier = Modifier
                                .fillMaxHeight()
                                .fillMaxWidth(progress)
                                .background(Gold)
                        )
                    }
                }
            }
        }
    }
}
